### PR TITLE
Homework Podstawy C++ password-check

### PR DESCRIPTION
**Pull request by rgruberski**
Authors: @rgruberski, @PawelYak

### CODE REVIEW

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @rgruberski

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @PawelYak

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @rgruberski

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @rgruberski

**Comment by ziobron** [comment]
⏱ 3 XP extra for @rgruberski for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @PawelYak

**Comment by ziobron** [comment]
⏱ 3 XP extra for @PawelYak for on-time delivery

File: homework/password-check/validation.cpp [hunk 1]
```diff
@@ -1,2 +1,59 @@
 #include "validation.hpp"
-// TODO: Put implementations here
\ No newline at end of file
+#include <algorithm>
+#include <cctype>
+#include <string>
+
+ErrorCode checkPasswordRules(std::string password) {
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:6]: Pass by const refrence. Try always avoid copy.

File: homework/password-check/validation.cpp [hunk 2]
```diff
@@ -1,2 +1,59 @@
...
+  }
+  if (!std::any_of(password.begin(), password.end(), isdigit)) {
+    return ErrorCode::PasswordNeedsAtLeastOneNumber;
+  }
+  if (!std::any_of(password.begin(), password.end(), ispunct)) {
+    return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+  }
+  if (!std::any_of(password.begin(), password.end(), isupper)) {
+    return ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter;
+  }
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:18]: better us none_of

File: homework/password-check/validation.cpp [hunk 3]
```diff
@@ -1,2 +1,59 @@
...
+  if (doPasswordsMatch(password, passwordConfirm)) {
+    return checkPasswordRules(password);
+  }
+  return ErrorCode::PasswordsDoNotMatch;
+}
+
+std::string getErrorMessage(ErrorCode errorCode) {
+
+  switch (errorCode) {
+
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:33]: Redundant new lines between switch and case

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**